### PR TITLE
Add toml support

### LINKED ISSUE #186: Add missing TOML file support
Add missing TOML file support as it is stated in the README.

### PR DESCRIPTION
**Pull request by gabor-boros**
**Reason for the change**

TOML file parsing support is a missing feature.

**Description**

Add TOML file parsing support.

**Code examples**

```python
TomlKeyExists(
    name="Ensure service descriptor has stack",
    path=Path("./service.toml"),
    key="stack",
    value="my-awesome-stack",
)
```

```python
TomlKeyNotExists(
    name="Ensure outdated_key is removed",
    path=Path("./service.toml"),
    key="outdated_key",
)
```

```python
TomlKeyRenamed(
    name="Ensure service descriptor has dependencies",
    path=Path("./service.toml"),
    key="development.depends_on",
    value="dependencies",
)
```

```python
TomlValueExists(
    name="Ensure service descriptor has dependencies",
    path=Path("./service.toml"),
    key="development.dependencies",
    value=["service1", "service2", "service3"],
)
```

```python
TomlValueNotExists(
    name="Remove decommissioned service from dependencies",
    path=Path("./service.toml"),
    key="development.dependencies",
    value="service4",
)
```

**Checklist**

- [x] Unit tests created/updated
- [x] Documentation extended/updated

**References**

resolves https://github.com/gabor-boros/hammurabi/issues/186

### CODE REVIEW

File: hammurabi/rules/toml.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+        >>> from pathlib import Path
+        >>> from hammurabi import Law, Pillar, TomlKeyExists
+        >>>
+        >>> example_law = Law(
+        >>>     name="Name of the law",
+        >>>     description="Well detailed description what this law does.",
+        >>>     rules=(
+        >>>         TomlKeyExists(
+        >>>             name="Ensure service descriptor has stack",
```
Comment (uvegla) [inline hammurabi/rules/toml.py:96]: Service descriptor? There are a couple of these below as well.

File: hammurabi/rules/toml.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        >>>
+        >>> pillar = Pillar()
+        >>> pillar.register(example_law)
+    """
+
+
+class TomlValueExists(DictValueExists, SingleDocumentTomlFileRule):
+    """
+    Ensure that the given key has the expected value(s). In case the key cannot
+    be found, a ``LookupError`` exception will be raised to stop the execution.
```
Comment (uvegla) [inline hammurabi/rules/toml.py:179]: Would it make sense to create the key if it does not exist? But then how does it differ from key exist check that also sets the value. I am not sure we need both. 🤔

**Review by uvegla — APPROVED** [review]

File: hammurabi/rules/toml.py (hunk 2 above)
Comment (gabor-boros) [reply hammurabi/rules/toml.py:179]: `TomlValueIsSet` is set could be used here (and for yaml/json as well)

Comment (gabor-boros) [reply hammurabi/rules/toml.py:179]: Should be part of https://github.com/gabor-boros/hammurabi/issues/174

**Pull request merged**